### PR TITLE
created the component

### PR DESCRIPTION
**Pull request by rishabhPlivo**

### CODE REVIEW

File: src/components/GroupAndSearchDropdown/index.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+export const GroupAndSearchDropdown = ({ options, onSearchChange, className, ...props }) => {
+  const [showDropdown, setShowDropdown] = useState(false)
+  const [value, setValue] = useState('')
+
+  const OptionsDropdown = () => {
+    return (
+      <div className="options-dropdown">
+        <div className="search-box">
+          <TextField type="search-box" placeholder="Search" onChange={onSearchChange} />
```
Comment (samuellawrentz) [inline src/components/GroupAndSearchDropdown/index.js:17]: How is the value cleared?
i.e what happens when you close and open the dropdown? Is the `searchValue` retained?

File: src/components/GroupAndSearchDropdown/GroupAndSearchDropdown.stories.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+export const Default = Template.bind({})
+Default.args = {
+  type: 'input',
+  size: 'default',
+  maxLength: 100,
+  placeholder: 'A simple text field',
+  options: [
+    {
+      title: 'System Variables',
+      children: ['Abcd', 'efgh', 'erekug', 'fhjdkg'],
```
Comment (samuellawerentz) [inline src/components/GroupAndSearchDropdown/GroupAndSearchDropdown.stories.js:27]: We should be having an object, which has `label` and `value`.

File: src/components/GroupAndSearchDropdown/groupAndSearchDropdown.scss [hunk 3]
```diff
@@ -0,0 +1,41 @@
+.contacto-icon--input-suffix-variable-dropdown {
+  position: absolute;
+  cursor: pointer;
+  font-size: 20px;
+  right: 0;
+  top: 0;
+  z-index: 1000;
```
Comment (samuellawerentz) [inline src/components/GroupAndSearchDropdown/groupAndSearchDropdown.scss:7]: We can make this 10. `1000` seems too much.

File: src/components/GroupAndSearchDropdown/index.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react'
+import { Dropdown } from 'antd'
+import PropTypes from 'prop-types'
+import { Icon } from '../Icon'
+import { Text } from '../Typography'
+import { TextField } from '../Textfield/index'
+import './groupAndSearchDropdown.scss'
+
+export const GroupAndSearchDropdown = ({ options, onSearchChange, className, ...props }) => {
```
Comment (samuellawerentz) [inline src/components/GroupAndSearchDropdown/index.js:9]: `onSearchChange` could be an internal function.

File: src/components/GroupAndSearchDropdown/index.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+              <Text type="captions_bold" color="gray-2">
+                {option?.title}
+              </Text>
+            </div>
+            {option?.children?.map((child, childIndex) => (
+              <div
+                className="group-option"
+                key={childIndex}
+                onClick={() => {
+                  setValue(child)
```
Comment (samuellawerentz) [inline src/components/GroupAndSearchDropdown/index.js:31]: on selection, we should append the selected value to the existing value. We should not clear it.

**Review by samuellawerentz — CHANGES_REQUESTED** [review]

**Pull request merged**